### DISCUSSION TITLE
Can we use `NavigationManager` inside a job?

### DISCUSSION THREAD

**Discussion by MrYossu** [quartznet/quartznet#1344] [Q&A]
I need to get the base URI for the web site that hosts my Quartz job. The usual way to do this is to use `NavigationManager.ToAbsoluteUri()`.

However, if I inject an instance of `NavigationManager` into a job, then it throws an exception "_System.InvalidOperationException' in Microsoft.AspNetCore.Components.dll 'RemoteNavigationManager' has not been initialized_" when it calls that method.

My guess is that the job is called outside of the context of an HTTP request, so whatever `NavigationManager` uses in that method isn't available (but that's just a guess, so please correct me if I'm wrong).

So how do I get the base URI from inside a job?
Thanks

**Comment by lahma** [comment]
Duplicate of https://stackoverflow.com/questions/69608235/how-do-i-inject-navigationmanager-in-a-quartz-net-job

**Comment by MrYossu** [comment]
@lahma It's only a duplicate in that I asked it there yesterday. However, it has hardly had any views, and given that the people here are more likely to be experts in Quartz, I thought I'd ask here.

Thanks anyway. Any idea of the answer to my question?

**Reply by kvantetore** [reply]
This does not really have anything to do with Quartz, and everything to do with AspNet/Blazor. I've tried to answer your question on SO
